tools/tf_env_collect.sh

```diff
@@ -52,84 +67,86 @@ platform.python_compiler(),
-platform.release(),
-platform.platform(),
-platform.linux_distribution(),
-platform.dist(),
-platform.mac_ver(),
-platform.uname(),
-platform.architecture(),
-platform.machine(),
-))
+PLATFORM_ENTRIES = [
+    ("os", "system"),
+    ("os kernel version", "version"),
+    ("os release version", "release"),
+    ("os platform", "platform"),
+    ("freedesktop os release", "freedesktop_os_release"),
+    ("mac version", "mac_ver"),
+    ("uname", "uname"),
+    ("architecture", "architecture"),
+    ("machine", "machine"),
+]
+
+for label, function_name in PLATFORM_ENTRIES:
+    if hasattr(platform, function_name):
+        function = getattr(platform, function_name)
+        result = function()  # Call the function
+        print(f"{label}: {result}")
+    else:
+        print(f"{label}: N/A")
 EOF
-${python_bin_path} /tmp/check_os.py 2>&1  >> ${OUTPUT_FILE}
 
-{
-  echo
-  echo '== are we in docker ============================================='
-  num=`cat /proc/1/cgroup | grep docker | wc -l`;
-  if [ $num -ge 1 ]; then
+
+  print_header 'are we in docker'
+  if grep -q docker /proc/1/cgroup; then
     echo "Yes"
   else
     echo "No"
   fi
   
-  echo
-  echo '== compiler ====================================================='
-  c++ --version 2>&1
-  
-  echo
-  echo '== check pips ==================================================='
-  pip list 2>&1 | grep "proto\|numpy\|tensorflow"
+  print_header 'c++ compiler'
+  if which c++; then
+    c++ --version 2>&1
+  else
+    echo "Not found"
+  fi
   
+  print_header 'check pips'
+  pip list 2>&1 | grep -E 'proto|numpy|tensorflow|tf_nightly'
   
-  echo
-  echo '== check for virtualenv ========================================='
-  ${python_bin_path} -c "import sys;print(hasattr(sys, \"real_prefix\"))"
   
-  echo
-  echo '== tensorflow import ============================================'
-} >> ${OUTPUT_FILE}
+  print_header 'check for virtualenv'
+  "${PYTHON_BIN_PATH}" <<EOF
+import sys
+
+if hasattr(sys, 'real_prefix') or (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix):
+    print("Running inside a virtual environment.")
+else:
+    print("Not running inside a virtual environment.")
+EOF
 
-cat <<EOF > /tmp/check_tf.py
+  print_header 'tensorflow import'
+
+  "${PYTHON_BIN_PATH}" <<EOF 2>&1
 import tensorflow as tf;
-print("tf.version.VERSION = %s" % tf.version.VERSION)
-print("tf.version.GIT_VERSION = %s" % tf.version.GIT_VERSION)
-print("tf.version.COMPILER_VERSION = %s" % tf.version.COMPILER_VERSION)
-with tf.Session() as sess:
-  print("Sanity check: %r" % sess.run(tf.constant([1,2,3])[:1]))
+print(f"""
+tf.version.VERSION = {tf.version.VERSION}
+tf.version.GIT_VERSION = {tf.version.GIT_VERSION}
+tf.version.COMPILER_VERSION = {tf.version.COMPILER_VERSION}
+""")
+print("Sanity check: %r" % tf.constant([1,2,3])[:1])
 EOF
-${python_bin_path} /tmp/check_tf.py 2>&1  >> ${OUTPUT_FILE}
 
-LD_DEBUG=libs ${python_bin_path} -c "import tensorflow"  2>>${OUTPUT_FILE} > /tmp/loadedlibs
```

Comment:
I had no idea what this code was trying to do, but LD_DEBUG logs are written to STDERR, so they were being put into the output file, and the loadedlibs file was blank every time. I assumed that this was trying to find libcudnn, so that's what it does now.